tokio/src/net/unix/listener.rs

```diff
@@ -96,21 +110,17 @@ impl UnixListener {
     /// # Examples
     ///
     /// ```rust,no_run
-    /// use std::error::Error;
-    ///
-    /// #[tokio::main]
-    /// async fn main() -> Result<(), Box<dyn Error>> {
-    ///     let tokio_listener = tokio::net::UnixListener::bind("127.0.0.1:0")?;
-    ///     let std_listener = tokio_listener.into_std()?;
-    ///     std_listener.set_nonblocking(false)?;
-    ///     Ok(())
-    /// }
+    /// # tokio::doctest_unix! {
+    /// let tokio_listener = tokio::net::UnixListener::bind("127.0.0.1:0")?;
+    /// let std_listener = tokio_listener.into_std()?;
+    /// std_listener.set_nonblocking(false)?;
+    /// # Ok(()) }
     /// ```
     ///
     /// [`tokio::net::UnixListener`]: UnixListener
-    /// [`std::os::unix::net::UnixListener`]: std::os::unix::net::UnixListener
-    /// [`set_nonblocking`]: fn@std::os::unix::net::UnixListener::set_nonblocking
-    pub fn into_std(self) -> io::Result<std::os::unix::net::UnixListener> {
+    /// [`std::os::unix::net::UnixListener`]: crate::os::unix::net::UnixListener
+    /// [`set_nonblocking`]: fn@crate::os::unix::net::UnixListener::set_nonblocking
+    pub fn into_std(self) -> io::Result<net::UnixListener> {
```

Comment:
Do these links work?

Reply:
Locally - yes. All though I really need to test this stuff on docs.rs to be sure.

The only one which doesn't work is the one which references signal_hook_registry, since that's a unix-conditional dependency:
![image](https://user-images.githubusercontent.com/111092/117573967-2d57fe80-b0db-11eb-9412-4b455bf3ce2e.png)

Reply:
It is worth noting that if you build the documentation on windows where `std::os::unix::net` (etc...) aren't available, these would instead take you to the [stub documentation](https://github.com/tokio-rs/tokio/pull/3768/files#diff-e5a5274f115c60e4d091329e873f1abaedfbd87232af91da36fe412882aff372).

Which is kinda... weird. But the alternative is to either live with broken links for the platform which the documentation is not being generated on, or rewrite all of them to directly reference std docs instead.